src/main/webapp/app/tasks/status-form/status-form.component.ts

```diff
@@ -0,0 +1,89 @@
+import { CdkDrag, CdkDragDrop, CdkDropList, moveItemInArray } from '@angular/cdk/drag-drop';
+import { ChangeDetectionStrategy, Component, Inject, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { v4 as uuid } from 'uuid';
+
+import { TaskService } from '../task.service';
+import { Task, TaskList } from '../task';
+import { DataHelper } from 'app/shared/helpers';
+import { Status } from '../enums';
+
+@Component({
+  selector: 'tiny-status-form',
+  templateUrl: './status-form.component.html',
+  styleUrls: ['./status-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class StatusFormComponent implements OnInit {
+
+  tasks$: Observable<TaskList[] | Task[]>;
+  public defaultTasksStatus = Status;
+
+  statusForm: FormGroup = new FormGroup({
+    name: new FormControl('', Validators.required)
+  });
+
+  constructor(@Inject('TaskService') private taskService: TaskService) { }
+
+  ngOnInit(): void {
+    this.tasks$ = this.taskService.getAll();
+  }
+
+  /**
+   * drag and drop event
+   * @param event CdkDragDrop
+   * @param taskList list of all tasks
+   */
+  drop(event: CdkDragDrop<string[]>, taskList: TaskList[]): void {
```

Comment:
i would see this functionality in the service tbh like "reorderStatus" or something (same goes for all taskList mutating functionality in this component).

Reply:
It is a copy-paste issue. I could change it to reorderStatus. I didn't think naming will be a problem in the interview. I have focused on logic.